service/spotify/api.js

```diff
@@ -8,86 +8,31 @@
 
 const Playlist = require("./playlist/playlist");
 const Album = require("./album/album");
-const SpotifyWebApi = require("spotify-web-api-node");
-var api = undefined;
-
-function init(callback) {
-  if (api == undefined || api == null) {
-    console.log("GetToken");
-    getApiToken(
-      function (obj) {
-        api = obj;
-        callback();
-      },
-      (err) => console.log(err)
-    );
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
-
-function getPlaylistName(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Playlist.Name(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
-
-function getPlaylistsUser(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Playlist.User(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
-
-function getPlaylistTracks(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Playlist.Tracks(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
+const Track = require("./track/track");
+const Artist = require("./artist/artist");
 
-function getPlaylist(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Playlist.Playlist(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
-
-function getAlbum(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Album.Album(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
-
-function getAlbumTracks(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Album.Tracks(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
+const SpotifyWebApi = require("spotify-web-api-node");
 
-function searchAlbums(query, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Album.Search(api, query, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
-}
+const Functions = {
+  playlistsUser: Playlist.User,
+  playlistTracks: Playlist.Tracks,
+  playlist: Playlist.Playlist,
+  album: Album.Album,
+  albumTracks: Album.Tracks,
+  albums: Album.Search,
+  artistAlbums: Album.Artist,
+  track: Track.Track,
+  tracks: Track.Search,
+  artist: Artist.Artist,
+  relatedArtists: Artist.RelatedArtists,
+  artistTopTracks: Artist.TopTracks,
+  artists: Artist.Search,
+};
 
-function getArtistAlbums(id, callbackOk, callbackFail) {
-  if (api != undefined && api != null) {
-    Album.Artist(api, id, callbackOk, callbackFail);
-  } else {
-    callbackFail("Invalid API Token!");
-  }
+function Get(funct, param, callbackOk, callbackFail) {
+  getApiToken(function (api) {
+    funct(api, param, callbackOk, callbackFail);
+  }, callbackFail);
```

Comment:
Uma coisa legal de pensar também é que as vezes cê tá usando arrow function e as vezes não. Seria legal manter um padrão no máximo que der.
```suggestion
  getApiToken(api => {
    funct(api, param, callbackOk, callbackFail);
  }, callbackFail);
```

Reply:
Tem horas que eu tava trocando entre o uso de um ou outro por causa do prettier kkk, que em um caso deixava mais bonito que o outro dependendo. Tinha hora que usar arrow function naquelas funções uma dentro da outra, ficava estranho de ler kk, mas é bom manter padrão mesmo